wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -145,23 +146,39 @@
    * <p>Name allocations are computed once and reused because some types may be needed when
    * generating other types.
    */
-  private final LoadingCache<MessageType, NameAllocator> nameAllocators
-      = CacheBuilder.newBuilder().build(new CacheLoader<MessageType, NameAllocator>() {
-    @Override public NameAllocator load(MessageType type) throws Exception {
+  private final LoadingCache<Type, NameAllocator> nameAllocators
+      = CacheBuilder.newBuilder().build(new CacheLoader<Type, NameAllocator>() {
+    @Override public NameAllocator load(Type type) throws Exception {
       NameAllocator nameAllocator = new NameAllocator();
-      nameAllocator.newName("serialVersionUID", "serialVersionUID");
-      nameAllocator.newName("ADAPTER", "ADAPTER");
-      nameAllocator.newName("MESSAGE_OPTIONS", "MESSAGE_OPTIONS");
-      if (emitAndroid) {
-        nameAllocator.newName("CREATOR", "CREATOR");
-      }
-      Set<String> collidingNames = collidingFieldNames(type.fieldsAndOneOfFields());
-      for (Field field : type.fieldsAndOneOfFields()) {
-        String suggestion = collidingNames.contains(field.name())
-            ? field.qualifiedName()
-            : field.name();
-        nameAllocator.newName(suggestion, field);
+
+      if (type instanceof MessageType) {
+        nameAllocator.newName("serialVersionUID", "serialVersionUID");
```

Comment:
Ah, right. It's for a one-shot name where you use the return value and don't care about it again.